code/modules/clothing/under/schoolgirl.dm

```diff
@@ -0,0 +1,137 @@
+/obj/item/clothing/under/schoolgirl
+	name = "blue schoolgirl uniform"
+	desc = "It's just like one of my Japanese animes!"
+	icon_state = "schoolgirl"
+	item_state = "schoolgirl"
+	item_color = "schoolgirl"
+	body_parts_covered = CHEST|GROIN|ARMS
+	fitted = FEMALE_UNIFORM_TOP
+	can_adjust = FALSE
+
+/obj/item/clothing/under/schoolgirl/red
+	name = "red schoolgirl uniform"
+	icon_state = "schoolgirlred"
+	item_state = "schoolgirlred"
+	item_color = "schoolgirlred"
+
+/obj/item/clothing/under/schoolgirl/green
+	name = "green schoolgirl uniform"
+	icon_state = "schoolgirlgreen"
+	item_state = "schoolgirlgreen"
+	item_color = "schoolgirlgreen"
+
+/obj/item/clothing/under/schoolgirl/orange
+	name = "orange schoolgirl uniform"
+	icon_state = "schoolgirlorange"
+	item_state = "schoolgirlorange"
+	item_color = "schoolgirlorange"
+
+/obj/item/clothing/under/schoolgirl/locked
+	name = "pink schoolgirl uniform"
+	desc = "It has a little lock inside. It's also super cute."
+	icon_state = "schoolgirlpink"
+	item_state = "schoolgirlpink"
+	item_color = "schoolgirlpink"
+	has_sensor = NO_SENSORS
+	sensor_mode = SENSOR_COORDS
+	random_sensor = FALSE
+	pocket_storage_component_path = /datum/component/storage/concrete/pockets/schoolgirl
+	custom_price = 100
+	var/locked = FALSE
+	var/perma = FALSE
+	var/mob/wearer = null
+	var/slaved = FALSE
+	var/mob/slaved_to = null
+
+/obj/item/clothing/under/schoolgirl/locked/equipped(mob/user, slot)
+	. = ..()
+	if(slot == SLOT_W_UNIFORM)
+		wearer = user
+		lock(user)
+
+/obj/item/clothing/under/schoolgirl/locked/dropped(mob/user)
+	. = ..()
+	wearer = null
+
+/obj/item/clothing/under/schoolgirl/locked/proc/toggle_lock(user)
+	locked = !locked
+	if(locked)
+		lock(user)
+	if(!locked)
+		unlock(user)

```

Comment:
This entire proc is bad. You are already switching the locked var in the lock/unlock procs which makes a lot more sense. Why do you also do it here?

Reply:
I was trying to clean up my code - originally I did the ENTIRE operation in that segmen - i moved it to allow for easier sorting of effects.